graph_def_editor/graph.py

```diff
@@ -1188,75 +1183,22 @@ def _decode_graph(graph_def):
   return output_map
 
 
-def _make_collection_defs(tf_g: tf.Graph) -> Iterable[
-  tf.MetaGraphDef.CollectionDefEntry]:
-  """
-  Convenience function to serialize all the collections in a TensorFlow graph.
+def _extract_collection_defs(meta_graph: tf.MetaGraphDef) -> Dict[
+  str, meta_graph_pb2.CollectionDef]:
 
-  Args:
-    tf_g: TensorFlow graph from which to harvest collections
-
-  Returns a list of `tf.MetaGraphDef.CollectionDefEntry` protobuf containing
-  the serialized contents of the collections.
-  """
-  ret = []
-  for collection_name in tf_g.collections:
+  collections = {}
+  for collection_name in meta_graph.collection_def:
     if type(collection_name) is not str:
       print("Skipping non-string collection name {}".format(collection_name))
       continue
-    collection_items = tf_g.get_collection(collection_name)
-    collection_proto = tf.MetaGraphDef.CollectionDefEntry()
-    collection_proto.key = collection_name
-    if len(collection_items) == 0:
-      continue
-    first_item_type = type(collection_items[0])
-    if issubclass(first_item_type, tf.Variable):
-      collection_type = tf.Variable
-    elif issubclass(first_item_type, tf.Operation):
-      collection_type = tf.Operation
-    elif issubclass(first_item_type, tf.Tensor):
-      collection_type = tf.Tensor
-    elif first_item_type.__name__ in ("WhileContext", "CondContext"):
-      print("Skipping collection {} of type {}.".format(
-        collection_name, first_item_type))
+    elif collection_name in (
+            "while_context", "cond_context", "saver", "queue_runners"):
+      print("Skipping collection {}".format(collection_name))
       # TODO(frreiss): Should we serialize WhileContexts or CondContexts?
       continue
-    else:
```

Comment:
This part of the diff doesn't look right. Was there a merge problem on this branch?

Reply:
I think the github diff is just messed up. It might be easier to view if you look at the function _extract_collection_defs in the file.